### ISSUE TITLE
Revisit #5954: Rule proposal: no-unsafe-readonly-mutable-assignment

### ISSUE THREAD

**Issue by jendrikw** [typescript-eslint/typescript-eslint#11422]
### Before You File a Proposal Please Confirm You Have Done The Following...

- [x] I have [searched for related issues](https://github.com/typescript-eslint/typescript-eslint/issues?q=is%3Aissue+label%3A%22enhancement%3A+new+plugin+rule%22) and found none that match my proposal.
- [x] I have searched the [current rule list](https://typescript-eslint.io/rules/#rules) and found no rules that match my proposal.
- [x] I have [read the FAQ](https://typescript-eslint.io/linting/troubleshooting) and my problem is not listed.

### My proposal is suitable for this project

- [x] My proposal specifically checks TypeScript syntax, or it proposes a check that requires type information to be accurate.
- [x] My proposal is not a "formatting rule"; meaning it does not just enforce how code is formatted (whitespace, brace placement, etc).
- [x] I believe my proposal would be useful to the broader TypeScript community (meaning it is not a niche proposal).

### Description

Since `--enforceReadonly` was never released (https://github.com/microsoft/TypeScript/pull/58296 is still open), it may be a good idea to think about implementing this.

### Fail Cases

```TypeScript
type MutableA = { a: string };
type ReadonlyA = { readonly a: string };

const mutate = (mut: MutableA): void => {
    mut.a = "whoops";
};

const readonlyA: ReadonlyA = { a: "readonly?" };
mutate(readonlyA);
```

### Pass Cases

```TypeScript
type MutableA = { a: string };
type ReadonlyA = { readonly a: string };

const read = (mut: ReadonlyA): void => {
    // No longer allowed:
    // mut.a = "whoops";
};

const readonlyA: ReadonlyA = { a: "readonly?" };
read(readonlyA);
```

### Additional Info

Other implementations:

- https://github.com/iliubinskii/eslint-plugin-misc/blob/master/src/typescript/base/no-unsafe-object-assignment.ts
- (eslint 8) https://github.com/danielnixon/eslint-plugin-total-functions/blob/master/src/rules/no-unsafe-mutable-readonly-assignment.ts

**Previous issue: #5954**

**Comment by bradzacher** [comment]
If there are two existing implementations and an open PR from Anders to add it to TS itself -- I'm not sure if we should move on this. I definitely agree there's value in this, but the existing implementations are probable good enough to cover the gap until the TS team merges the PR?

cc @typescript-eslint/triage-team

**Comment by kirkwaiblinger** [comment]
Agreed. In addition to the points you've made, this _seems_ like it would be high complexity to implement (`no-unsafe-*`-like), which is especially hard to justify for a possibly-temporary state of affairs.

**Comment by jendrikw** [comment]
@bradzacher one has many deprecated dependencies (https://github.com/iliubinskii/eslint-plugin-misc/issues/7) and the other one is for eslint 8 only, so effectively, there's no usable version at the moment. But I can totally understand that the implementation effort isn't worth it.

**Comment by JoshuaKGoldberg** [comment]
Indeed. As much as we would love to have a good implementation in typescript-eslint, we don't  have the maintenance bandwidth to tackle all of the important initiatives we want to as it is. A large tricky new rule like this one needs a lot of care, experimentation, and iteration. The best case here would be for the community to take on that work. Closing out as wontfix for us, with the hopes someone else takes on this valuable work. Cheers!
